### DISCUSSION TITLE
Is there any way to not auto-focus the output_panel when running tests?

### DISCUSSION THREAD

**Discussion by Irubataru** [nvim-neotest/neotest#216] [Q&A] [answered]
When I run my tests it automatically focuses the output_panel when the tests are done (assuming I have failures), is there a way not to do that?

Also, this seems to make it so the `output` briefly flash before it focus is shifted and it closes. I am not 100% sure about that. I tried to make a minimal setup to test if there are setting or plugin collisions, but when I run my minimal setup it for some reason skips the tests if I try to run them.

Nvim version:
```
NVIM v0.9.0-dev-1094+ga0292b4e5f
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/clang -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fno-common -fdiagnostics-color=always -fstack-protector-strong -DNVIM_MSGPACK_HAS_FLOAT32 -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DMIN_LOG_LEVEL=3 -DNVIM_UNIBI_HAS_VAR_FROM -I/usr/include/luajit-2.1 -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/home/alegle/.cache/paru/clone/neovim-git/src/build/src/nvim/auto -I/home/alegle/.cache/paru/clone/neovim-git/src/build/include -I/home/alegle/.cache/paru/clone/neovim-git/src/build/cmake.config -I/home/alegle/.cache/paru/clone/neovim-git/src/neovim/src -I/usr/include -I/usr/include -I/usr/include

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

Attempt at minimal setup:
```lua
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  -- bootstrap lazy.nvim
  -- stylua: ignore
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", "--branch=stable",
      lazypath })
end
vim.opt.rtp:prepend(vim.env.LAZY or lazypath)

require("lazy").setup({
  spec = {
    {
      "nvim-neotest/neotest",
      dependencies = {
        { "nvim-lua/plenary.nvim" },
        { "nvim-treesitter/nvim-treesitter" },
        { "antoinemadec/FixCursorHold.nvim" },

        -- Adapters
        { "nvim-neotest/neotest-python" },
      },
      config = function()
        require("neotest").setup({
          adapters = {
            require("neotest-python"),
          },
        })
      end,
    },
  },
  defaults = {
    lazy = false,
  },
})
```

**Comment by rcarriga** [comment, marked as answer]
The output panel shouldn't auto-focus.

>  automatically focuses the output_panel when the tests are done (assuming I have failures),

As in it only focuses on the window when there are failures?

> but when I run my minimal setup it for some reason skips the tests if I try to run them.

Is there any output from the tests? 

There is a minimal setup provided in the bug report template if you can't get your minimal version working?

**Reply by Irubataru** [reply]
> As in it only focuses on the window when there are failures?

Yes, it auto-focus the output window when there are failures, but not when there are none. It also seems to auto-focus the second time I do `require("neotest").run.run()`. But again, these are the types of strange problems where getting the minimal config to work would really help... But I can't...

> Is there any output from the tests?

No, it says "No tests found", and if I open the summary they are there, but skipped for some reason (with my version of the minimal config).

![image](https://user-images.githubusercontent.com/5611071/222679191-b31d75fb-5b5d-43ad-bde7-a30653095343.png)

It does work in my full setup, so I don't know what is missing. Running pytest outside of nvim finds the tests.

> There is a minimal setup provided in the bug report template if you can't get your minimal version working?

For some reason when I run the minimal setup it doesn't find the tests at all, I do not understand. I just have a dummy python project I made to test neotest, and I use pytest in it.

**Reply by Irubataru** [reply]
Here is a recording of the behaviour I see

https://asciinema.org/a/rJ3j71qY1YvYGWNhlLh15cY4v

By the way, I also have this other issue that I have not figured out what is, and I cannot reliably reproduce

https://asciinema.org/a/54wtFk45Nhhar87Ec9LUhjDbT

**Reply by Irubataru** [reply]
I finally managed to get the minimal configs working. In my own I had forgot to run `TSInstall python`, and in yours I added the adapter later but forgot to rerun `:PackerSync`... sorry! However, I still see the same behaviour, I even got both of the issues

https://asciinema.org/a/HATZaycuAhUJ0dhayIHl1rY6B

And here is one without the weird bottom-line resize

https://asciinema.org/a/XcLEdGTlsGfot9Q68SaxDhfKa

_Edit:_ I wanted to check if the weird thing that happens where it resizes was a tmux thing, but it isn't. I also tried xterm instead of alactritty, but I got the same issue there as well. I also tired running it in neovim v0.8, but I get the same behaviour (both bugs).

**Reply by rcarriga** [reply]
Ah this is not the output panel. It is the quickfix list being filled by the `quickfix` consumer. You can set it to not open by default
```lua
  neotest.setup({
    quickfix = {
      open = false,
    },
    ...
```
See `:help neotest.Config.quickfix`

**Reply by Irubataru** [reply]
Ah, perfect, that worked, thanks! I have no idea why I thought it was the output. When I ran `require("neotest").output.open()` something strange might have happened that made me confused, or maybe I was generally confused.

By the way, do you have any idea what happens with the strange resize? The one from the first minimal example? I assume it won't happen if the qf doesn't open, but it shouldn't behave like that regardless.
